From a7512b437a10c904264d247dbcdf051427443e11 Mon Sep 17 00:00:00 2001
From: Arjan van de Ven <arjan.van.de.ven@intel.com>
Date: Thu, 1 Sep 2022 17:27:50 +0000
Subject: [PATCH] allow for non-versioned delta files

---
 src/swupd_lib/delta.c | 18 +++++++++++-------
 1 file changed, 11 insertions(+), 7 deletions(-)

diff --git a/src/swupd_lib/delta.c b/src/swupd_lib/delta.c
index dc075ae6..eaa09932 100644
--- a/src/swupd_lib/delta.c
+++ b/src/swupd_lib/delta.c
@@ -79,18 +79,22 @@ static bool check_delta_filename(const char *delta_name, char *from, char *to)
 {
 	/* Delta files have the form [FROM_VERSION]-[TO_VERSION]-[FROM_HASH]-[TO_HASH]. */
 	const char *s = delta_name;
+	/* Note: SWUPD_HASH_LEN accounts for the NUL-terminator after the hash. */
+	const size_t hash_len = SWUPD_HASH_LEN - 1;
 
 	/* Ignore versions, deltas will be used based on their hashes only. */
-	for (int i = 0; i < 2; i++) {
-		s = strchr(s, '-');
-		if (!s) {
-			return false;
+	/* As of September 2022, the server no longer puts the version prefix in place, but handle */
+	/* both cases */
+	if ( str_len(s) > hash_len * 2 + 1) {
+		for (int i = 0; i < 2; i++) {
+			s = strchr(s, '-');
+			if (!s) {
+				return false;
+			}
+			s++;
 		}
-		s++;
 	}
 
-	/* Note: SWUPD_HASH_LEN accounts for the NUL-terminator after the hash. */
-	const size_t hash_len = SWUPD_HASH_LEN - 1;
 	if (str_len(s) != (hash_len * 2 + 1)) {
 		return false;
 	}
-- 
2.37.2

