diff --git a/src/swupd_lib/packs.c b/src/swupd_lib/packs.c
index 35607a36..e9165f9d 100644
--- a/src/swupd_lib/packs.c
+++ b/src/swupd_lib/packs.c
@@ -43,6 +43,9 @@
 */
 #define MAX_XFER 1
 
+const char * largest_pack = NULL;
+unsigned long largest_pack_size = 0;
+
 struct pack_data {
 	char *url;
 	char *filename;
@@ -115,11 +118,18 @@ static bool download_error(enum download_status status, void *data)
 static bool download_successful(void *data)
 {
 	struct pack_data *pack_data = data;
+	struct stat st;
 
 	if (!pack_data) {
 		return false;
 	}
 
+	if (stat(pack_data->filename, &st) == 0) {
+		if (st.st_size > largest_pack_size) {
+			largest_pack_size = st.st_size;
+			largest_pack = strdup(pack_data->module);
+		}
+	}
 	return finalize_pack_download(pack_data->module, pack_data->newversion, pack_data->filename) == 0;
 }
 
diff --git a/src/swupd_lib/stats.c b/src/swupd_lib/stats.c
index 7f5ff2ad..9e91c997 100644
--- a/src/swupd_lib/stats.c
+++ b/src/swupd_lib/stats.c
@@ -28,6 +28,9 @@
 #include "swupd.h"
 
 int swupd_stats[8];
+extern const char * largest_pack;
+extern unsigned long largest_pack_size;
+
 
 void print_statistics(int version1, int version2)
 {
@@ -39,5 +42,8 @@ void print_statistics(int version1, int version2)
 	info("    changed files     : %i\n", swupd_stats[2]);
 	info("    new files         : %i\n", swupd_stats[0]);
 	info("    deleted files     : %i\n", swupd_stats[1]);
+	if (largest_pack) {
+		info("    largest pack      : %s %iKb\n", largest_pack, largest_pack_size/1024);
+	}
 	info("\n");
 }
diff --git a/src/lib/strings.h b/src/lib/strings.h
index 2658ac71..df72f9e0 100644
--- a/src/lib/strings.h
+++ b/src/lib/strings.h
@@ -24,7 +24,7 @@ extern "C" {
 /**
  * helper to use strncmp() instead of strcmp
  */
-#define str_cmp(_str1, _str2) strncmp(_str1, _str2, INT64_MAX)
+#define str_cmp(_str1, _str2) strcmp(_str1, _str2)
 
 /**
  * Check if str starts with the prefix.
diff --git a/src/lib/strings.h b/src/lib/strings.h
index 2658ac71..12e33705 100644
--- a/src/lib/strings.h
+++ b/src/lib/strings.h
@@ -19,7 +19,7 @@ extern "C" {
 /**
  * Helper to use strnlen() instead of strlen.
  */
-#define str_len(_str) strnlen(_str, INT64_MAX)
+#define str_len(_str) strlen(_str)
 
 /**
  * helper to use strncmp() instead of strcmp
